modules/rest/src/main/java/org/apache/ignite/internal/rest/routes/SimpleRouter.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.rest.routes;
+
+import io.netty.handler.codec.http.HttpRequest;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.apache.ignite.rest.Route;
+import org.apache.ignite.rest.Routes;
+
+/**
+ * Dispatcher of http requests.
+ *
+ * <p>Example:
+ * <pre>
+ * {@code
+ * var router = new SimpleRouter();
+ * router.get("/user", (req, resp) -> {
+ *     resp.status(HttpResponseStatus.OK);
+ * });
+ * }
+ * </pre>
+ */
+public class SimpleRouter implements Router, Routes {
+    /** Routes. */
+    private final List<Route> routes;
+
+    /**
+     * Creates a new router with the given list of {@code routes}.
+     *
+     * @param routes Routes.
+     */
+    public SimpleRouter(List<Route> routes) {
+        this.routes = List.copyOf(routes);
```

Comment:
1. This constructor is not used anywhere.
2. `List.copyOf` creates an immutable list, `addRoute` which breaks the contract of this class (e.g. `addRoute` will fail)

Reply:
Removed the constructor, it's easy to add it back if we need it in the future